### PR TITLE
fix #2410 Add serialized wrappers for Sinks.Empty/Sinks.One

### LINKED ISSUE #2410: Add serialized wrappers for SinkOne and Sink Empty
Similarly to what has been done for SinkMany, create serialized wrappers for SinkOne and SinkEmpty.

### PR DESCRIPTION
**Pull request by aneveu**

### CODE REVIEW

File: reactor-core/src/jcstress/java/reactor/core/publisher/SinkOneSerializedStressTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import reactor.util.context.Context;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.LockSupport;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+
+public class SinkOneSerializedStressTest {
+
+	final SinkOneSerializedStressTest.TargetSink<Object> stressSink = new SinkOneSerializedStressTest.TargetSink<>();
```
Comment (bsideup) [inline reactor-core/src/jcstress/java/reactor/core/publisher/SinkOneSerializedStressTest.java:27]:
```suggestion
	final TargetSink<Object> stressSink = new TargetSink<>();
```

File: reactor-core/src/jcstress/java/reactor/core/publisher/SinkManySerializedStressTest.java [hunk 2]
```diff
@@ -45,7 +45,7 @@
 	@Outcome(id = {"FAIL_NON_SERIALIZED, OK, 1"}, expect = ACCEPTABLE, desc = "second wins")
 	@Outcome(id = {"OK, OK, 2"}, expect = ACCEPTABLE, desc = "one after another")
 	@State
-	public static class TryEmitNextStressTest extends SerializedManySinkStressTest {
+	public static class TryEmitNextManySerializedStressTest extends SinkManySerializedStressTest {
```
Comment (bsideup) [inline reactor-core/src/jcstress/java/reactor/core/publisher/SinkManySerializedStressTest.java:48]: I am wondering why IDEA renamed this class :D I think we should rename it back to keep the inner class names shorter

**Review by simonbasle — CHANGES_REQUESTED** [review]
I think we should wait for #2418 before merging this, which could then modify `Sinks.one()` and `Sinks.empty()` to return the serialized versions. At the same time, a review of (prod?) usage of both sinks would be in order (we probably need the unsafe versions inside operators)

File: reactor-core/src/jcstress/java/reactor/core/publisher/SinkManySerializedStressTest.java (hunk 2 above)
Comment (simonbasle) [reply reactor-core/src/jcstress/java/reactor/core/publisher/SinkManySerializedStressTest.java:48]: to clarify, this should indeed be renamed back to `TryEmitNextStressTest`

File: reactor-core/src/jcstress/java/reactor/core/publisher/SinkManySerializedStressTest.java [hunk 3]
```diff
@@ -69,7 +69,7 @@ public void arbiter(LLI_Result r) {
 	@Outcome(id = {"FAIL_TERMINATED, OK, 1"}, expect = ACCEPTABLE, desc = "onNext after onComplete")
 	@Outcome(id = {"OK, OK, 1"}, expect = ACCEPTABLE, desc = "onComplete after onNext")
 	@State
-	public static class TerminatedVsOnNextStressTest extends SerializedManySinkStressTest {
+	public static class TerminatedVsOnNextManySerializedStressTest extends SinkManySerializedStressTest {
```
Comment (simonbasle) [inline reactor-core/src/jcstress/java/reactor/core/publisher/SinkManySerializedStressTest.java:72]: same as in above @bsideup comment: rename back to `TerminatedVsOnNextStressTest`

**Review by simonbasle — CHANGES_REQUESTED** [review]

**Review by bsideup — APPROVED** [review]

**Review by simonbasle — APPROVED** [review]

**Pull request merged**